test/currencyConvertor.ts

```diff
@@ -1,73 +1,175 @@
+      params: [impersonatedAccount],
+    });
+    await hre.network.provider.request({
+      method: "hardhat_setBalance",
+      params: [
+        impersonatedAccount,
+        "0x1000000000000000000",
+      ],
+    });
+    const signer = await ethers.getSigner(impersonatedAccount);
 
-    zeroExExchangeWrapper = await deployContract(
-      signers[0],
-      ZeroExExchangeWrapperArtifact,
-    );
+    // deploy contracts
     starkware = await deployContract(
-      signers[0],
+      signer,
       StarkwareArtifact,
-    );
-    usdc = await deployContract(
-      signers[0],
-      UsdcContractArtifact,
-    )
-
-    await usdc.mint(
-      zeroExExchangeWrapper.address,
-      1000e6,
-    )
-    await usdc.mint(
-      starkware.address,
-      1000e6,
-    )
-
+      [uniswapAddress],
+    ) as MockStarkware;
     currencyConvertor = await deployContract(
-      signers[0],
+      signer,
       CurrencyConvertorArtifact,
       [
         starkware.address,
-        usdc.address, // placeholder for USDC
-        156,
+        uniswapAddress,
+        '1',
       ],
+    ) as CurrencyConvertor;
+    zeroExExchangeWrapper = await deployContract(
+      signer,
+      ZeroExExchangeArtifact,
+    ) as ZeroExExchangeWrapper;
+
+    // get ERC20 contracts
+    const usdcContract = new ethers.Contract(
+    usdcAddres,
+    tokenAbi,
+    signer,
     );
 
-    await usdc.mint(
-      signers[0].address,
-      1000e6,
-    )
+    console.log(await usdcContract.symbol());
 
-    await usdc.approve(currencyConvertor.address, 1e10);
+    // approve ERC20 contracts
+    await usdcContract.approve(
+    currencyConvertor.address,
+    1000e6,
+    );
   });
+
   describe("deposit", async () => {
-    it("deposit USDT to USDC", async () => {
-      await currencyConvertor.deposit(
-        usdc.address,
-        100,
+    it("deposit USDC to Uniswap", async () => {
+      const zeroExTransaction: any = await axiosRequest({
+        method: 'GET',
+        url: generateQueryPath(
+          'https://api.0x.org/swap/v1/quote',
+          {
+            sellToken: usdcAddres,
+            buyToken: uniswapAddress,
+            sellAmount: '1',
+            slippagePercentage: 1.0,
+          },
+        ),
+      });
+
+      const tx = await currencyConvertor.deposit(
+        usdcAddres,
+        '1000',
         zeroExExchangeWrapper.address,
-        100,
-        1,
-        0x4,
+        '10000', // starkKey
+        '100000', // positionId
+        encode(zeroExTransaction.to, zeroExTransaction.data),
       );
+
+      const blocks = await tx.wait();
+      const events = _.chain(blocks.events!)
+      .filter((e) => e.event === 'LogConvertedDeposit')
+      .value();
+
+      const event = events[0];
+      expect(event.args?.tokenFromAmount.toString()).to.equal('1000');
+      expect(event.args?.tokenFrom.toLowerCase()).to.equal(usdcAddres);
+    });
+
+    it("deposit USDC to Uniswap with invalid call data address", async () => {
```

Comment:
deposit USDC to Uniswap? This sounds like you're depositing USDC into a Uniswap pool 🤔